pkg/volume/csi/csi_plugin.go

```diff
@@ -440,9 +440,9 @@ func (p *csiPlugin) NewMounter(
 
 	if err := saveVolumeData(dataDir, volDataFileName, volData); err != nil {
 		klog.Error(log("failed to save volume info data: %v", err))
-		if err := os.RemoveAll(dataDir); err != nil {
-			klog.Error(log("failed to remove dir after error [%s]: %v", dataDir, err))
-			return nil, err
```

Comment:
Good catch, however, would it make sense to return `err` here (and below in NewBlockVolumeMapper)? Because that's why setup failed.